packages/astro/src/vite-plugin-astro/index.ts

```diff
@@ -216,6 +216,13 @@ export default function astro({ config, logging }: AstroPluginOptions): vite.Plu
 				return {
 					code: `${code}${SUFFIX}`,
 					map,
+					meta: {
+						vite: {
+							// Setting this vite metadata to `ts` causes Vite to resolve .js
+							// extensions to .ts files.
+							lang: 'ts'
+						}
+					}
```

Comment:
Nice! Do we also need this in `vite-plugin-markdown`?

Reply:
Probably, will add a test for that.

Reply:
It does now